static/js/src/advantage/users/AccountUsers.tsx

```diff
@@ -1,35 +1,68 @@
 import React, { useState } from "react";
-
-import { User, Users, OrganisationName } from "./types";
+import { useQueryClient, useMutation, useQuery } from "react-query";
+import {
+  User,
+  Users,
+  OrganisationName,
+  NewUserValues,
+  UserRole,
+} from "./types";
 import Organisation from "./components/Organisation";
 import AddNewUser from "./components/AddNewUser/AddNewUser";
 import TableView from "./components/TableView/TableView";
 import DeleteConfirmationModal from "./components/DeleteConfirmationModal/DeleteConfirmationModal";
+import { requestAddUser, requestDeleteUser, requestUpdateUser } from "./api";
 
-type Props = {
+export type AccountUsersProps = {
   organisationName: OrganisationName;
+  accountId: string;
   users: Users;
 };
 
-const AccountUsers = ({ organisationName, users }: Props) => {
+const AccountUsers = ({
+  accountId,
+  organisationName,
+  users,
+}: AccountUsersProps) => {
   const [hasNewUserSuccessMessage, setHasNewUserSuccessMessage] = useState(
     false
   );
   const [
     hasUserDeletedSuccessMessage,
     setHasUserDeletedSuccessMessage,
   ] = useState(false);
-  const handleAddNewUser = (value: string) => {
-    return Promise.resolve(value).then(() => {
+
+  const queryClient = useQueryClient();
+
+  const userAddMutation = useMutation(
+    (user: NewUserValues) => requestAddUser({ accountId, ...user }),
+    { onSuccess: () => queryClient.invalidateQueries("accountUsers") }
+  );
+
+  const userDeleteMutation = useMutation(
+    (email: string) => requestDeleteUser({ accountId, email }),
+    { onSuccess: () => queryClient.invalidateQueries("accountUsers") }
+  );
+
+  const userUpdateMutation = useMutation(
+    ({ email, role }: { email: string; role: UserRole }) =>
+      requestUpdateUser({ accountId, email, role }),
+    {
+      onSuccess: () => queryClient.invalidateQueries("accountUsers"),
+    }
+  );
+
+  const handleAddNewUser = (user: NewUserValues) =>
+    userAddMutation.mutateAsync(user).then(() => {
       setHasNewUserSuccessMessage(true);
```

Comment:
Why do this here instead of the `onSuccess`| of `userAddMutation`?

Reply:
I needed the `mutateAsync` so that components can respond locally on resolve/rejection wIthout having to pass state updates via props.

You can't use `onSuccess` with `mutateAsync` AFAIK https://react-query.tanstack.com/guides/mutations#promises 

Using `.then()` just seemed cleaner if I'm awaiting the promise in the component anyway.

This is the AddNewUserForm that's waiting for the promise:

https://github.com/canonical-web-and-design/ubuntu.com/blob/bb427360984a4f4a91bc58fe0fd8f63a44a4995c/static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx#L36-L43